homeassistant/components/weatherflow/config_flow.py

```diff
@@ -0,0 +1,77 @@
+"""Config flow for WeatherFlow."""
+from __future__ import annotations
+
+import asyncio
+from asyncio import Future
+from asyncio.exceptions import CancelledError
+from typing import Any
+
+from pyweatherflowudp.client import EVENT_DEVICE_DISCOVERED, WeatherFlowListener
+from pyweatherflowudp.errors import AddressInUseError, EndpointError, ListenerError
+
+from homeassistant import config_entries
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import (
+    DOMAIN,
+    ERROR_MSG_ADDRESS_IN_USE,
+    ERROR_MSG_CANNOT_CONNECT,
+    ERROR_MSG_NO_DEVICE_FOUND,
+)
+
+
+async def _async_can_discover_devices() -> bool:
+    """Return if there are devices that can be discovered."""
+    future_event: Future[None] = asyncio.get_running_loop().create_future()
+
+    @callback
+    def _async_found(_):
+        """Handle a discovered device - only need to do this once so."""
+
+        if not future_event.done():
+            future_event.set_result(None)
+
+    async with WeatherFlowListener() as client, asyncio.timeout(10):
+        try:
+            client.on(EVENT_DEVICE_DISCOVERED, _async_found)
+            await future_event
+        except asyncio.TimeoutError:
+            return False
+        except (EndpointError, CancelledError):
+            raise ListenerError
```

Comment:
Its a bit strange to convert this into `ListenerError`.  I'll post a suggested diff shortly if I can get the wifi working well enough at the airport

Reply:
```diff
diff --git a/homeassistant/components/weatherflow/config_flow.py b/homeassistant/components/weatherflow/config_flow.py
index 3b92871898d..3a108ec658d 100644
--- a/homeassistant/components/weatherflow/config_flow.py
+++ b/homeassistant/components/weatherflow/config_flow.py
@@ -28,7 +28,6 @@ async def _async_can_discover_devices() -> bool:
     @callback
     def _async_found(_):
         """Handle a discovered device - only need to do this once so."""
-
         if not future_event.done():
             future_event.set_result(None)
 
@@ -38,8 +37,6 @@ async def _async_can_discover_devices() -> bool:
             await future_event
         except asyncio.TimeoutError:
             return False
-        except (EndpointError, CancelledError):
-            raise ListenerError
 
     return True
 
@@ -65,7 +62,7 @@ class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
             found = await _async_can_discover_devices()            
         except AddressInUseError:
             errors["base"] = ERROR_MSG_ADDRESS_IN_USE
-        except ListenerError:
+        except (ListenerError, EndpointError, CancelledError):
             errors["base"] = ERROR_MSG_CANNOT_CONNECT
 
         if not found and not errors:
```